### PR TITLE
feat: Add `temporary` argument to `duckdb_read_csv()`

### PR DESCRIPTION
**Pull request by ThomasSoeiro**
Here is a try for #142

@krlmlr: [You suggested](https://github.com/duckdb/duckdb-r/issues/142#issuecomment-2294792558) to modify the SQL and to use `tbl_function()`.

I do not understand why it is needed. Cannot we just pass temporary to `dbCreateTable()` like I did? Can you please clarify?

Warning: this example will overwrite `test.db` and `iris.csv` files on your disk if present.

```r
library(duckdb)
con <- dbConnect(duckdb(), dbdir = "test.db")

write.csv(iris, "iris.csv", row.names = FALSE)

duckdb_read_csv(con, "iris_temp_false", "iris.csv")
duckdb_read_csv(con, "iris_temp_true", "iris.csv", temporary = TRUE)

dbListTables(con)
# [1] "iris_temp_false" "iris_temp_true" 

dbReadTable(con, "iris_temp_false") |> head()
#   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
# 1          5.1         3.5          1.4         0.2  setosa
# 2          4.9         3.0          1.4         0.2  setosa
# 3          4.7         3.2          1.3         0.2  setosa
# 4          4.6         3.1          1.5         0.2  setosa
# 5          5.0         3.6          1.4         0.2  setosa
# 6          5.4         3.9          1.7         0.4  setosa

dbReadTable(con, "iris_temp_true") |> head()
#   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
# 1          5.1         3.5          1.4         0.2  setosa
# 2          4.9         3.0          1.4         0.2  setosa
# 3          4.7         3.2          1.3         0.2  setosa
# 4          4.6         3.1          1.5         0.2  setosa
# 5          5.0         3.6          1.4         0.2  setosa
# 6          5.4         3.9          1.7         0.4  setosa

dbDisconnect(con)
```

Check that the temporary table is gone:

```
con <- dbConnect(duckdb::duckdb(), dbdir = "test.db")

dbListTables(con)
# [1] "iris_temp_false"

dbDisconnect(con)
```

### CODE REVIEW

File: R/csv.R [hunk 1]
```diff
@@ -29,8 +29,8 @@
 #' dbReadTable(con, "data")
 #'
 #' dbDisconnect(con)
-duckdb_read_csv <- function(conn, name, files, header = TRUE, na.strings = "", nrow.check = 500,
-                            delim = ",", quote = "\"", col.names = NULL, lower.case.names = FALSE, sep = delim, transaction = TRUE, ...) {
+duckdb_read_csv <- function(conn, name, files, ..., header = TRUE, na.strings = "", nrow.check = 500,
```
Comment (krlmlr) [inline R/csv.R:32]: We'd want to check that the ellipsis is actually empty. This is touching the same file, but the move of the ellipsis and the check could also be a separate PR.

**Review by krlmlr — COMMENTED** [review]
Thanks, I was confused and thought there's a `COPY TEMPORARY ... FROM` syntax for disambiguation. There doesn't seem to be one: https://duckdb.org/docs/sql/statements/copy.html#copy--from .

Can you please also add a test?

File: R/csv.R (hunk 1 above)
Comment (ThomasSoeiro) [reply R/csv.R:32]: I have added a warning. Is it what you wanted?

**Comment by ThomasSoeiro** [comment]
@krlmlr 

Since yesterday, I've received about 75 email notifications about vendor workflow failures, and counting.
How can I stop this?
Thanks!

**Comment by krlmlr** [comment]
Thanks. I'm sorry that the vendoring workflow gives you headaches. The easiest is to turn off actions in your fork. The alternative is to fix the workflow so that it doesn't run in forks, there's a similar pattern in my other actions. But this PR uses the main branch of your fork, so fixing there will disturb the PR.

**Comment by krlmlr** [comment]
Thanks!

**Comment by ThomasSoeiro** [comment]
Hi @krlmlr,
Do we need anything else so that this PR can be merged?
Thanks!

**Comment by krlmlr** [comment]
Thanks for the nudge. I thought I had turned on auto-merge for this PR.
